x/tokenomics/keeper/settle_session_accounting.go

```diff
@@ -150,24 +150,37 @@ func (k Keeper) SettleSessionAccounting(
 		// goes "into debt". Need to design a way to handle this when we implement
 		// probabilistic proofs and add all the parameter logic. Do we touch the application balance?
 		// Do we just let it go into debt? Do we penalize the application? Do we unstake it? Etc...
-		settlementAmt = sdk.NewCoin("upokt", math.Int(application.Stake.Amount))
-		settlementAmtuPOKT = sdk.NewCoins(settlementAmt)
-		// TODO_BLOCKER(@Olshansk): The application should be immediately unstaked at this point in time
+		settlementAmt = cosmostypes.NewCoin("upokt", math.Int(application.Stake.Amount))
+		settlementAmtuPOKT = cosmostypes.NewCoins(settlementAmt)
+
+		applicationOverservicedEvent := &tokenomicstypes.EventApplicationOverserviced{
+			ApplicationAddr: applicationAddress.String(),
+			ExpectedBurn:    &settlementAmt,
```

Comment:
`settlementAmt` has already be overridden by `application.Stake.Amount` at this point.

Reply:
Can you say more, I'm not quite following. :thinking:

Reply:
`settlementAmt` is overridden at line `153` to `application.Sake.Amount` so it's no longer the expected amount to burn.

At this point in code, `settlementAmt` and `application.Stake` are the same. What we should do is to `settlementAmt` to a `expectedSettleemntAmt` before reassigning `settlementAmt` to `application.Stake.Amount` (`L153`) then use the new variable in `ExpectedBurn`.